src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/routing/AuditMessageRouter.java

```diff
@@ -51,7 +52,7 @@
 
 	public AuditMessageRouter(final Settings settings, final Client clientProvider, ThreadPool threadPool, final Path configPath) {
 		this.sinkProvider = new SinkProvider(settings, clientProvider, threadPool, configPath);
-		this.storagePool = new AsyncStoragePool(settings);
+		this.storagePool = new AsyncStoragePool(AsyncStoragePoolConfig.getConfig(settings));
```

Comment:
Is it necessary to change the signature of the `AsyncStoragePool` constructor? Can it continue to accept `Settings` and create `AsyncStoragePoolConfig` inside?

Reply:
I believe it is a good idea to pass only whats needed to a class.

Reply:
Yes, but on the other side, now `AuditMessageRouter` is exposed to `Settings`, `AsyncStoragePool` and in addition to `AsyncStoragePoolConfig`.

Reply:
Those also will undergo changes.

Reply:
All required dependencies will be injected for all the classes

Reply:
I really want to keep scope small for PRs as previous PRs went un-attended for weeks.